ff/src/fields/models/quadratic_extension.rs

```diff
@@ -192,11 +192,17 @@ impl<P: QuadExtParameters> One for QuadExtField<P> {
 
 impl<P: QuadExtParameters> Field for QuadExtField<P> {
     type BasePrimeField = P::BasePrimeField;
+    type BasePrimeFieldArr = Vec<Self::BasePrimeField>;
 
     fn extension_degree() -> u64 {
         2 * P::BaseField::extension_degree()
     }
 
+    fn to_base_prime_field_elements(&self) -> Self::BasePrimeFieldArr {
+	self.c0.to_base_prime_field_elements().as_ref().into_iter().map(|ci| *ci).chain(self.c1.to_base_prime_field_elements().as_ref().into_iter().map(|ci| *ci)).collect::<Vec<Self::BasePrimeField>>()
```

Comment:
I still don't know why I needed to dually dereference (once with into_iter and then using map). Maybe some one with better understanding of rust iterator can re-rewrite better.

Reply:
I think you can just change it to `iter`?